torchvision/prototype/transforms/_misc.py

```diff
@@ -32,6 +35,68 @@ def extra_repr(self) -> str:
         return ", ".join(extras)
 
 
+class LinearTransformation(Transform):
+
+    _transformed_types: Tuple[Type, ...] = (torch.Tensor, features.Image, PIL.Image.Image)
+
+    def __init__(self, transformation_matrix: torch.Tensor, mean_vector: torch.Tensor):
+        super().__init__()
+        if transformation_matrix.size(0) != transformation_matrix.size(1):
+            raise ValueError(
+                "transformation_matrix should be square. Got "
+                f"{tuple(transformation_matrix.size())} rectangular matrix."
+            )
+
+        if mean_vector.size(0) != transformation_matrix.size(0):
+            raise ValueError(
+                f"mean_vector should have the same length {mean_vector.size(0)}"
+                f" as any one of the dimensions of the transformation_matrix [{tuple(transformation_matrix.size())}]"
+            )
+
+        if transformation_matrix.device != mean_vector.device:
+            raise ValueError(
+                f"Input tensors should be on the same device. Got {transformation_matrix.device} and {mean_vector.device}"
+            )
+
+        self.transformation_matrix = transformation_matrix
+        self.mean_vector = mean_vector
+
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+
+        pil_mode = None
+        if isinstance(inpt, PIL.Image.Image):
+            pil_mode = inpt.mode
+            inpt = F.to_image_tensor(inpt).contiguous()
+
+        shape = inpt.shape
+        n = shape[-3] * shape[-2] * shape[-1]
+        if n != self.transformation_matrix.shape[0]:
+            raise ValueError(
+                "Input tensor and transformation matrix have incompatible shape."
+                + f"[{shape[-3]} x {shape[-2]} x {shape[-1]}] != "
+                + f"{self.transformation_matrix.shape[0]}"
+            )
```

Comment:
Shouldn't this happen in `_get_params`?

Reply:
There are no params to generate in this transform. I think moving this code to `_get_params` and returning empty dict will look a bit strange, IMO

Reply:
Well, the `n` is parameter, no?

Reply:
No, `n` is not a parameter, we only check if input is compatible with `self.transformation_matrix` and that's it. `n` wont change for the inputs.

Reply:
`n` is the product of the number of channels, height, and width. Since it 

> wont change for the inputs.

we should have it in `_get_params` with the 

```py
image = query_image(sample)
num_channels, height, width = F.get_image_dimensions(image)
n = num_channels * height * width

....

return dict(n=n)
```

idiom.

Reply:
@vfdev-5 I think that's worth adding here to be consistent with other implementations. For tiny changes like this and the TODO below (https://github.com/pytorch/vision/pull/6458#discussion_r951206479) it might be worth doing them on the same PR. Otherwise we just increasing the TODO list which makes it harder to track and easier to miss.